diff --git a/src/components/organisms/ListInsideClusterAndNs/ListInsideClusterAndNs.tsx b/src/components/organisms/ListInsideClusterAndNs/ListInsideClusterAndNs.tsx
index ac56e5f..c6e2350 100644
--- a/src/components/organisms/ListInsideClusterAndNs/ListInsideClusterAndNs.tsx
+++ b/src/components/organisms/ListInsideClusterAndNs/ListInsideClusterAndNs.tsx
@@ -1,6 +1,6 @@
 import React, { FC, useState } from 'react'
 import { Button, Alert, Spin, Typography } from 'antd'
-import { filterSelectOptions, Spacer, useBuiltinResources, useApiResources } from '@prorobotech/openapi-k8s-toolkit'
+import { filterSelectOptions, Spacer, useApiResources } from '@prorobotech/openapi-k8s-toolkit'
 import { useNavigate } from 'react-router-dom'
 import { useSelector, useDispatch } from 'react-redux'
 import { RootState } from 'store/store'
@@ -11,6 +11,11 @@ import {
   CUSTOM_NAMESPACE_API_RESOURCE_RESOURCE_NAME,
 } from 'constants/customizationApiGroupAndVersion'
 import { Styled } from './styled'
+import {
+  BASE_PROJECTS_API_GROUP,
+  BASE_PROJECTS_VERSION,
+  BASE_PROJECTS_RESOURCE_NAME,
+} from 'constants/customizationApiGroupAndVersion'
 
 export const ListInsideClusterAndNs: FC = () => {
   const clusterList = useSelector((state: RootState) => state.clusterList.clusterList)
@@ -33,9 +38,11 @@ export const ListInsideClusterAndNs: FC = () => {
     typeof CUSTOM_NAMESPACE_API_RESOURCE_RESOURCE_NAME === 'string' &&
     CUSTOM_NAMESPACE_API_RESOURCE_RESOURCE_NAME.length > 0
 
-  const namespacesData = useBuiltinResources({
+  const namespacesData = useApiResources({
     clusterName: selectedCluster || '',
-    typeName: 'namespaces',
+    apiGroup: BASE_PROJECTS_API_GROUP,
+    apiVersion: BASE_PROJECTS_VERSION,
+    typeName: BASE_PROJECTS_RESOURCE_NAME,
     limit: null,
     isEnabled: selectedCluster !== undefined && !isCustomNamespaceResource,
   })
diff --git a/src/hooks/useNavSelectorInside.ts b/src/hooks/useNavSelectorInside.ts
index 5736e2b..1ec0f71 100644
--- a/src/hooks/useNavSelectorInside.ts
+++ b/src/hooks/useNavSelectorInside.ts
@@ -1,6 +1,11 @@
-import { TClusterList, TSingleResource, useBuiltinResources } from '@prorobotech/openapi-k8s-toolkit'
+import { TClusterList, TSingleResource, useApiResources } from '@prorobotech/openapi-k8s-toolkit'
 import { useSelector } from 'react-redux'
 import { RootState } from 'store/store'
+import {
+  BASE_PROJECTS_API_GROUP,
+  BASE_PROJECTS_VERSION,
+  BASE_PROJECTS_RESOURCE_NAME,
+} from 'constants/customizationApiGroupAndVersion'
 
 const mappedClusterToOptionInSidebar = ({ name }: TClusterList[number]): { value: string; label: string } => ({
   value: name,
@@ -15,9 +20,11 @@ const mappedNamespaceToOptionInSidebar = ({ metadata }: TSingleResource): { valu
 export const useNavSelectorInside = (clusterName?: string) => {
   const clusterList = useSelector((state: RootState) => state.clusterList.clusterList)
 
-  const { data: namespaces } = useBuiltinResources({
+  const { data: namespaces } = useApiResources({
     clusterName: clusterName || '',
-    typeName: 'namespaces',
+    apiGroup: BASE_PROJECTS_API_GROUP,
+    apiVersion: BASE_PROJECTS_VERSION,
+    typeName: BASE_PROJECTS_RESOURCE_NAME,
     limit: null,
     isEnabled: Boolean(clusterName),
   })
diff --git a/src/utils/getBacklink.ts b/src/utils/getBacklink.ts
index a862354..f24e2bc 100644
--- a/src/utils/getBacklink.ts
+++ b/src/utils/getBacklink.ts
@@ -28,7 +28,7 @@ export const getFormsBackLink = ({
   }
 
   if (namespacesMode) {
-    return `${baseprefix}/${clusterName}/builtin-table/namespaces`
+    return `${baseprefix}/${clusterName}/api-table/core.cozystack.io/v1alpha1/tenantnamespaces`
   }
 
   if (possibleProject) {
@@ -64,7 +64,7 @@ export const getTablesBackLink = ({
   }
 
   if (namespacesMode) {
-    return `${baseprefix}/${clusterName}/builtin-table/namespaces`
+    return `${baseprefix}/${clusterName}/api-table/core.cozystack.io/v1alpha1/tenantnamespaces`
   }
 
   if (possibleProject) {
