diff --git a/controller/pkg/bucketclaim/bucketclaim.go b/controller/pkg/bucketclaim/bucketclaim.go
index 2f4d565e..2addcee8 100644
--- a/controller/pkg/bucketclaim/bucketclaim.go
+++ b/controller/pkg/bucketclaim/bucketclaim.go
@@ -165,7 +165,7 @@ func (b *BucketClaimListener) provisionBucketClaimOperation(ctx context.Context,
 			return b.recordError(inputBucketClaim, v1.EventTypeWarning, v1alpha1.FailedCreateBucket, err)
 		}
 
-		bucketName = bucketClassName + string(bucketClaim.ObjectMeta.UID)
+		bucketName = fmt.Sprintf("bucket-%s", bucketClaim.ObjectMeta.UID)
 
 		// create bucket
 		bucket := &v1alpha1.Bucket{}
